.scripts/lib.sh

```diff
@@ -0,0 +1,92 @@
+#
+# Copyright (c) 2021 Snowplow Analytics Ltd. All rights reserved.
+#
+# This program is licensed to you under the Apache License Version 2.0,
+# and you may not use this file except in compliance with the Apache License Version 2.0.
+# You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the Apache License Version 2.0 is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+#
+
+
+# -----------------------------------------------------------------------------
+#  FUNCTIONS & PROCEDURES
+# -----------------------------------------------------------------------------
+
+function die() {
+    echo "$@" 1>&2 ; exit 1;
+}
+
+# Given a tag and a module name,
+#   if the tag is a release tag for module, echoes a tag
+#   else returns 1
+# Uses shopt and sets locale, so runs in subshell
+function is_release_tag() (
+    [ "$#" -eq 2 ] || die "2 arguments required, $# provided to 'func:is_release_tag'"
+    local __tag="$1"
+    local __module="$2"
+
+    # For extended glob patterns and ensure [0-9] range
+    shopt -s extglob
+    LC_ALL=C
+
+    case "${__module}" in
```

Comment:
I think we will need to update this to add a JavaScript tracker specific rule to it. 

With v3, we will end up with a `3.0.0` tag but also a tag for each package (that will happen automatically by our publishing tool) like `@snowplow/browser-tracker_v3.0.0`. We only need the 3.0.0 style tags here, not the NPM package tags.
If you'd like an example, it will end up something like this repo: https://github.com/paulboocock/rush-playground